modules/lib/src/main/resources/assets/js/v6/features/store/dialogs/deleteDialog.store.ts

```diff
@@ -0,0 +1,364 @@
+import {showError} from '@enonic/lib-admin-ui/notify/MessageBus';
+import {NotifyManager} from '@enonic/lib-admin-ui/notify/NotifyManager';
+import {i18n} from '@enonic/lib-admin-ui/util/Messages';
+import {computed, map} from 'nanostores';
+import {ContentSummaryAndCompareStatus} from '../../../../app/content/ContentSummaryAndCompareStatus';
+import {ContentId} from '../../../../app/content/ContentId';
+import {ContentServerChangeItem} from '../../../../app/event/ContentServerChangeItem';
+import {ArchiveContentRequest} from '../../../../app/resource/ArchiveContentRequest';
+import {ContentSummaryAndCompareStatusFetcher} from '../../../../app/resource/ContentSummaryAndCompareStatusFetcher';
+import {DeleteContentRequest} from '../../../../app/resource/DeleteContentRequest';
+import {ResolveDeleteRequest} from '../../../../app/resource/ResolveDeleteRequest';
+import {$contentArchived, $contentDeleted} from '../socket.store';
+import {createDebounce} from '../../utils/timing/createDebounce';
+import {hasContentIdInIds} from '../../utils/cms/content/ids';
+
+export type DeleteAction = 'archive' | 'delete';
+
+//
+// * Store state
+//
+
+type DeleteDialogStore = {
+    open: boolean;
+    loading: boolean;
+    failed: boolean;
+    submitting: boolean;
+    items: ContentSummaryAndCompareStatus[];
+    dependants: ContentSummaryAndCompareStatus[];
+    inboundTargets: ContentId[];
+    inboundIgnored: boolean;
+    archiveMessage: string;
+    pendingAction?: DeleteAction;
+    pendingIds: string[];
+    pendingTotal: number;
+    pendingPrimaryName?: string;
+    yesCallback?: () => void;
+    noCallback?: () => void;
+};
```

Comment:
Store is too complex. You are mixing base properties with some properties to depend your actions and responses on. Functions that you use should be pure and do not depend on external state, only on the arguments.